### PR TITLE
Unit test improvements

### PR DESCRIPTION
**Pull request by guusdk**
This mostly migrates unit tests in Openfire's code base from JUnit 4 to 5. The PR includes some other minor unit test improvements. Please refer to the commit messages for more details.

### CODE REVIEW

File: xmppserver/pom.xml [hunk 1]
```diff
@@ -462,15 +462,33 @@
         </dependency>
         <!-- Test Scope -->
         <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
-            <version>4.13.1</version>
+            <groupId>org.junit.jupiter</groupId>
```
Comment (GregDThomas) [inline xmppserver/pom.xml:465]: I'd be very tempted to use the JUnit BOM for this. 
```
        <dependency>
            <groupId>org.junit</groupId>
            <artifactId>junit-bom</artifactId>
            <version>5.9.3</version>
            <type>pom</type>
            <scope>import</scope>
        </dependency>
```
This removes the need to set the version for any of your JUnit dependencies - so next time you want to update the version, change it in the one place.

**Review by GregDThomas — APPROVED** [review]
I've not the detail of all the files you've changed, but those I have been through all look good.

File: xmppserver/pom.xml [hunk 2]
```diff
@@ -462,15 +462,33 @@
...
+            <groupId>org.junit.vintage</groupId>
+            <artifactId>junit-vintage-engine</artifactId>
+            <version>5.9.3</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.mockito</groupId>
-            <artifactId>mockito-inline</artifactId>
-            <version>4.9.0</version>
+            <artifactId>mockito-core</artifactId>
```
Comment (GregDThomas) [inline xmppserver/pom.xml:484]: There's a mockito BOM too
```
<dependency>
    <groupId>org.mockito</groupId>
    <artifactId>mockito-bom</artifactId>
    <version>5.4.0</version> // or 5.3.1 if there's a good reason to use the older one
    <type>pom</type>
    <scope>import</scope>
</dependency>
```

**Comment by guusdk** [comment]
Thanks @GregDThomas! I've pushed a commit with the changes that you suggested.

**Review by Fishbowler — APPROVED** [review]
Wow Guus! This is an awesome improvement, and a real slice of work!

I'd be remiss if I didn't include a "unit tests pass" joke here, but unfortunately I can't think of one.

**Pull request merged**